src/clients/RTorrent.ts

```diff
@@ -297,6 +397,60 @@ export default class RTorrent implements TorrentClient {
 		}
 	}
 
+	async getAllTorrents(): Promise<GenericTorrentInfo[]> {
+		const hashes = await this.methodCallP<string[]>("download_list", []);
+		type ReturnType = string[][] | Fault[];
+		let response: ReturnType;
+		try {
+			response = await this.methodCallP<ReturnType>("system.multicall", [
+				hashes.map((hash) => {
+					return {
+						methodName: "d.custom1",
+						params: [hash],
+					};
+				}),
+			]);
+		} catch (e) {
+			logger.error({
+				Label: Label.RTORRENT,
+				message: "Failed to get torrent info for all torrents",
+			});
+			logger.debug(e);
+			return [];
```

Comment:
is this going to be a leaky blocklist?

Reply:
I'm not sure what we can do about it. But if anything, I think that should be handled by the caller.

Reply:
I still think this should maybe throw or return an ErrResult, but probably not a huge deal. We can move forward.